### PR TITLE
Improve phys perf

### PR DESCRIPTION
**Pull request by cerati**

### CODE REVIEW

**Comment by cerati** [comment]
benchmarks: https://cerati.web.cern.ch/cerati/PR271/

File: Config.h [hunk 1]
```diff
@@ -354,8 +355,8 @@ namespace Config
   constexpr float track_ptlow = 0.9;
 
   // sorting config (bonus,penalty)
-  constexpr float validHitBonus_ = 7.5;//cmssw bonus = 2.5
-  constexpr float missingHitPenalty_ = 5.0;//cmssw penalty = 20
+  constexpr float validHitBonus_ = 10;//XX;//7.5;//20.;//7.5;//cmssw bonus = 2.5
+  constexpr float missingHitPenalty_ = 10;//XX;// 5.0;//7.5;//5.0;//cmssw penalty = 20
```
Comment (areinsvo) [inline Config.h]: @cerati could you clean up the comments on these lines?

File: Geoms/CMS-2017.cc [hunk 2]
```diff
@@ -17,9 +17,10 @@ namespace
 
     Config::useCMSGeom       = true;
     Config::nlayers_per_seed = 4;
-    Config::maxCandsPerSeed  = 6;  // GC said 3 is enough ???
-    Config::maxHolesPerCand  = 3; // = 12;
-    Config::chi2Cut          = 15.0; // = 30.0;
+    Config::maxCandsPerSeed  = 5;//6;//4;  // GC said 3 is enough ???
+    Config::maxHolesPerCand  = 4;//XX; // = 12;
+    Config::maxConsecHoles   = 1;//XX; // = 12;
+    Config::chi2Cut          = 30;//XX; //15.0; // = 30.0;
```
Comment (areinsvo) [inline Geoms/CMS-2017.cc]: @cerati could you clean up the comments on these lines?

File: Track.h [hunk 3]
```diff
@@ -35,7 +35,8 @@ struct IdxChi2List
   unsigned int seedtype; // seed type idx (used for sorting: 0 = not set; 1 = high pT central seeds; 2 = low pT endcap seeds; 3 = all other seeds)
   float pt;   // pt (used for sorting)
   float chi2;   // total chi2 (used for sorting)
-  int score; // score used for candidate ranking
+  float score; // score used for candidate ranking
+  /* int score; // score used for candidate ranking */
```
Comment (areinsvo) [inline Track.h]: @cerati is the commented out line still useful to keep around?

File: Track.h [hunk 4]
```diff
@@ -577,7 +580,23 @@ inline float getScoreCalc(const unsigned int seedtype,
...
   score_ = Config::validHitBonus_*nfoundhits - Config::missingHitPenalty_*nmisshits - chi2;
-  if(pt<0.9f && seedtype==2) score_ -= 0.5f*(Config::validHitBonus_)*nfoundhits;
+  if(seedtype==2) {
+    score_ -= 0.5f*(Config::validHitBonus_)*nfoundhits;
+  }
+  if (seedtype==2 || seedtype==3) {
+    if (nfoundhits<=8) {
+      score_ -= 0.06f*(Config::validHitBonus_)*nfoundhits;
+    } else if (nfoundhits>12) {
+      score_ += 0.08f*(Config::validHitBonus_)*nfoundhits;
```
Comment (areinsvo) [inline Track.h:586]: I'm surprised these relatively small changes to the hit bonus for short and long tracks make a difference

File: mkFit/HitStructures.h [hunk 5]
```diff
@@ -332,7 +332,8 @@ inline bool sortByScoreTrackCand(const TrackCand & cand1, const TrackCand & cand
   return cand1.score() > cand2.score();
 }
 
-inline int getScoreCand(const TrackCand& cand1)
+/* inline int getScoreCand(const TrackCand& cand1) */
```
Comment (areinsvo) [inline mkFit/HitStructures.h]: Is it useful to keep the commented out line here? @cerati

**Comment by areinsvo** [comment]
The changes in this pull request were summarized in the slides from Giuseppe [here](https://indico.cern.ch/event/895997/contributions/3777798/attachments/2015179/3368140/mkFit-plots-v2.pdf)

@cerati, I have a couple comments after looking at the benchmarks. Do you understand the striking new shape of the nHits distributions after your changes? I'm comparing [before](https://mkortela.web.cern.ch/mkortela/tracking/mkfit/PR/PR265/SIMVAL_MTV_SEED/nHits/SKL-SP_CMSSW_TTbar_PU50_allmatch_nHits_build_pt0p0_SIMVALSEED.png) and [after](https://cerati.web.cern.ch/cerati/PR271/SIMVAL_MTV_SEED/nHits/SKL-SP_CMSSW_TTbar_PU50_bestmatch_nHits_build_pt0p0_SIMVALSEED.png) 

After:
SKL-SP_CMSSW_TTbar_PU50_bestmatch_nHits_build_pt0p0_SIMVALSEED


Also, the efficiency for short tracks (nlayers < 10) with pt > 0.9 seems to get slightly worse: Comparing [before](https://mkortela.web.cern.ch/mkortela/tracking/mkfit/PR/PR265/SIMVAL_MTV_SEED/SKL-SP_CMSSW_TTbar_PU50_eff_nLayers_build_pt0p9_SIMVALSEED.png) and [after](https://cerati.web.cern.ch/cerati/PR271/SIMVAL_MTV_SEED/SKL-SP_CMSSW_TTbar_PU50_eff_nLayers_build_pt0p9_SIMVALSEED.png), but maybe this isn't such a big deal.

**Comment by cerati** [comment]
The shape is due to the change in validHitBonus_:
`if (nfoundhits<=8) {
      score_ -= 0.06f*(Config::validHitBonus_)*nfoundhits;
    } else if (nfoundhits>12) {
      score_ += 0.08f*(Config::validHitBonus_)*nfoundhits;
}`
So although they seem small modifications they have a noticeable impact.

I did notice the efficiency loss in that region, I guess it's a price we need to pay for all the other improvements. 

Thanks for the comments to the code, I will make another commit later today.

**Comment by cerati** [comment]
Commit made.

**Pull request merged**